### ISSUE TITLE
Screen readers are not able to read the PIN input screen.

### ISSUE THREAD

**Issue by jleach** [bcgov/bc-wallet-mobile#754]
When rereading the PIN, the screen reader reads the PIN as a single number (ex. 1 0 0 0 0 1 will read one hundred thousand and one). It should read each individual digit.

- Put spaces in-between each number
- Strip out spacing before submitting

Ref: #551

**Comment by jleach** [comment]
This happens because the splitting of digits is just astetics. Under the hood its just a text field that splits characters up for display. To fix this, we will need to find a better replacement component, write our own, or contribute a fix back to the main project. Regardless of the choice its going to be some effort.

**Comment by jleach** [comment]
Talked this over at backlog refinement today. We're going to test to see if we can add spaces to the digits in the underlying text field (and strip them out) to force the screen reader to recite each digit separately.

**Comment by jleach** [comment]
@bryce-mcmath Able to see what's up with this one now that we have the new PIN screen?

**Comment by jeznorth** [comment]
@jleach safe to assume this is effecting both IOS / Android devices?

**Comment by jleach** [comment]
@jeznorth Hmm. It's a bit old and I don't recall.

**Comment by bryce-mcmath** [comment]
@jeznorth just confirmed this issue still exists on iOS. Haven't tested Android but I'm fairly certain it will be the same there as well. While inputting the numbers, it reads them out individually, but when you focus on the input field after entering them, it reads them as one number (2222 is read as two thousand two hundred twenty two). I think Jason's suggested approach (some kind of character stripping) would likely work

**Comment by knguyenBC** [comment]
QC has a similar issue regarding the PIN input screen where the screen reader is reading the PIN even when it's hidden. They may be able to tackle this ticket at the same time.

**Comment by jeznorth** [comment]
I'm making an assumption we only want to hear the digits on this field (e.g. 1, 2, 3, 4). Testing on IOS 18.4.1, using the pin '1234' it still reads out 'one-thousand two-hundred thirty-four' BEFORE sounding out the digits independently.

**Comment by bryce-mcmath** [comment]
Going to quickly fix the remaining iOS-specific part of this issue as I have a physical device to test on. Great work by Matt so far.

**Comment by bryce-mcmath** [comment]
So this was a journey of ups and downs. Up, the PIN is properly read aloud on iOS. Down, it then also reads it as one massive number. Up, there is a prop we can add to prevent the value itself (the massive number) from being read called `secureTextEntry`. Down, this prop causes weird behaviour with backspace as noted [here](https://github.com/facebook/react-native/pull/45964). Up, there's another prop called `accessibilityValue` that can override the default VoiceOver behaviour of reading the value. Down, it only started working properly in [0.80.0-rc.5](https://github.com/facebook/react-native/commit/0c25f19d3944e866fcbb6a09da6a55878f739742)

In summary, I think it's good enough as is for now. The PIN is read aloud properly, it just unfortunately also reads the large number as well on iOS. We could patch RN to include the new fix but I don't think that's worth the time investment and general hairiness of patching RN.
